plugins/lang/python/python_plugin_context.cc

```diff
@@ -214,7 +214,16 @@ _sys.stderr = catctOutErr
 
 )foo"; //this is python code to redirect stdouts/stderr
 
-    PyRun_SimpleString(stdOutErr.c_str()); //invoke code to redirect
+    int res = PyRun_SimpleString(stdOutErr.c_str()); //invoke code to redirect
+    if (res) {
+        cerr << "error injecting logging code: " << endl;
+        PyErr_Print(); //make python print any errors, unfortunately to console
+        cerr << "MLDB will now raise an exception" << endl;
+        throw HttpReturnException
+            (500, "Couldn't inject Python code (see error message on console). "
+             "Have you installed python_dependenies (json and datetime) and "
```

Comment:
But if Python isn't initialized properly (eg, bad virtualenv) they can't be found and this segfaults.